driftbase/match_placements.py

```diff
@@ -0,0 +1,539 @@
+import logging
+import json
+import datetime
+import typing
+import uuid
+from flask import g
+from driftbase.models.db import Match
+from driftbase import flexmatch, lobbies
+from driftbase.lobbies import InvalidRequestException, NotFoundException, UnauthorizedException, _post_lobby_event_to_members, _get_lobby_member_player_ids, _get_lobby_key, _get_lobby_host_player_id, _get_player_lobby_key, _GenericLock, _LockedLobby
+
+from driftbase.resources.match_placements import TIER_DEFAULTS
+
+"""
+Only lobby matches with GameLift provider supported at the time of writing!!!
+"""
+
+# TODO: Not use protected/private functions in lobbies module
+
+log = logging.getLogger(__name__)
+
+def get_player_match_placement(player_id: int, expected_match_placement_id: typing.Optional[str] = None) -> dict:
+    placement = None
+    with _GenericLock(_get_player_match_placement_key(player_id)) as player_match_placement_lock:
+        placement_id = player_match_placement_lock.value
+
+        if not placement_id:
+            log.info(f"Player '{player_id}' attempted to fetch a match placement without having a match placement")
+            message = f"Match placement {expected_match_placement_id} not found" if expected_match_placement_id else "No match placement found"
+            raise NotFoundException(message)
+
+        if expected_match_placement_id and expected_match_placement_id != placement_id:
+            log.warning(f"Player '{player_id}' attempted to fetch match placement '{expected_match_placement_id}', but the player didn't issue the match placement")
+            raise UnauthorizedException(f"You don't have permission to access match placement {expected_match_placement_id}")
+
+        with _JsonLock(_get_match_placement_key(placement_id)) as match_placement_lock:
```

Comment:
per discussion, look into using redis watch instead of locks where you find yourself nesting locks.